### PR TITLE
Add local storage to issue count

### PR DESCRIPTION
**Pull request by Steven-Pingel**
Adding local storage to reduce the GH api calls.  Data is considered stale after 24 hours.

storage module inspired by : http://visualstudiomagazine.com/articles/2011/11/01/caching-in-on-html5-local-storage.aspx

### CODE REVIEW

File: javascripts/main.js [hunk 1]
```diff
@@ -47,19 +73,26 @@
...
-          , message = rateLimitReset ? 'GitHub rate limit met. Reset at ' + rateLimitReset.toLocaleTimeString() + '.' :
-                      'Could not get issue count from GitHub: ' + ((jqXHR.responseJSON && jqXHR.responseJSON.message) || errorThrown) + '.';
-        count.html('?');
-        count.attr('title', message);
-      });
+    count = $('<span class="count"><img src="images/octocat-spinner-32.gif" /></span>').appendTo(a);
+    var cached = storage.get(gh[1]);
+    if (cached && cached.date >= (new Date() - 1000 * 60 * 60 * 24)) {
+        count.html(cached.count)
+    } else {
```
Comment (dahlbyk) [inline javascripts/main.js]: Instead of `if`/`else` can we just `return` within the `if` to avoid the extra level of indentation?

**Comment by dahlbyk** [comment]
Added a `.gitattributes` in the parent PR - your changes use CRLF vs LF in the rest of the file. Can you rebase and normalize?

**Comment by dahlbyk** [comment]
Also added `.editorconfig` to parent PR - please conform to 2-space indentation. :smiling_imp: 

whitespace everywhere

File: javascripts/main.js [hunk 2]
```diff
@@ -34,6 +34,32 @@
     });
   });
 
+   var storage = function(global) {
```
Comment (dahlbyk) [inline javascripts/main.js]: Best practice to wrap IIFE in parens: `(function(global) { ... })(window)`

File: javascripts/main.js [hunk 3]
```diff
@@ -47,19 +73,26 @@
...
-        var rateLimited = jqXHR.getResponseHeader('X-RateLimit-Remaining') === '0'
-          , rateLimitReset = rateLimited && new Date(1000 * +jqXHR.getResponseHeader('X-RateLimit-Reset'))
-          , message = rateLimitReset ? 'GitHub rate limit met. Reset at ' + rateLimitReset.toLocaleTimeString() + '.' :
-                      'Could not get issue count from GitHub: ' + ((jqXHR.responseJSON && jqXHR.responseJSON.message) || errorThrown) + '.';
-        count.html('?');
-        count.attr('title', message);
-      });
+    count = $('<span class="count"><img src="images/octocat-spinner-32.gif" /></span>').appendTo(a);
+    var cached = storage.get(gh[1]);
+    if (cached && cached.date >= (new Date() - 1000 * 60 * 60 * 24)) {
```
Comment (dahlbyk) [inline javascripts/main.js]: I'm never seeing the cache hit. Looks like `cached.date` is a `string`?

Comment (Steven-Pingel) [inline javascripts/main.js]: I needed to convert the date string to an actual date :)  

Testing this helped me figure out how to set conditional breakpoints in ChromeDevTools.

**Comment by Steven-Pingel** [comment]
Should be hitting the cache now.  Thanks @dahlbyk!
cachehit

**Pull request merged**